### PR TITLE
WIP:Go 1.16 update to facilitate Apple Silicon builds

### PR DESCRIPTION
**Pull request by arlaneenalra**
This is a work in progress MR that updates Deps.Cloud to build with go 1.16 and produce an Apple silicon compatible cli tool.

### CODE REVIEW

File: Makefile
Comment (mjpitz) [inline Makefile:8]: ahhh... forgot about that... woops

File: Makefile
Comment (mjpitz) [inline Makefile:110]: I think there's a line earlier in this file that does this same cd && npm install .. that target could depend on this one.

File: DEVELOPING.md
Comment (mjpitz) [inline DEVELOPING.md:48]: nit: one sentence per line

Very rarely do we use a target width pattern in the project. I've found that it helps me identify run-ons and then I don't need to re-adjust things when there are edits.

Also, your first sentence here has `are are used`. `are used` is passive voice. I'd probably frame it more as `These images provide the foundation for building the rest of the application.`

`s/docker/container/` in text.

**Comment by mjpitz** [comment]
All in all this change looks good. I ran through a few smoke tests last night on a 1.16 built version and things appear to work fine. If we merge this in, I'm thinking of putting it in with the `0.3.0` release. We can do a few release candidate versions before doing the official `0.3.0`.

File: Makefile
Comment (arlaneenalra) [reply Makefile:110]: I see https://github.com/depscloud/depscloud/blob/main/Makefile#L17 ... It probably should, and this should probably be using install instead of ci. I'll update.

File: Makefile
Comment (arlaneenalra) [inline Makefile:21]: dang it..

Comment (mjpitz) [reply Makefile]: oh... yeah. Eventually, I shouldn't need the phony target here. Once I get the paths between modules PR merged, I'm planning on moving `deps`, `extractor`, `gateway`, `indexer`, and `tracker` all under a single `services` directory. But for now... :-/

File: DEVELOPING.md
Comment (arlaneenalra) [reply DEVELOPING.md:48]: And this is why I need an editor... one moment and I'll correct this.

Comment (arlaneenalra) [reply DEVELOPING.md:48]: If I'm reading this right, you're using a sentence per line?

File: Makefile
Comment (arlaneenalra) [reply Makefile]: Makes sense. the Dang it was specifically about the errant `L`...

**Review by mjpitz — APPROVED** [review]

**Comment by mjpitz** [comment]
Oh! @arlaneenalra we're missing the line from the `go.mod` file.

File: DEVELOPING.md
Comment (mjpitz) [reply DEVELOPING.md]: Yeah, it's a lot easier to edit later on if something needs to be added to a sentence.

**Comment by arlaneenalra** [comment]
Just seeing your reply, what line are we missing?

**Comment by arlaneenalra** [comment]
Ah, I see. That will be fixed shortly.

**Review by mjpitz — APPROVED** [review]

**Comment by mjpitz** [comment]
@arlaneenalra is this good to go? If so, I'll merge and get a new version out in the next couple of days.

**Comment by arlaneenalra** [comment]
Well, it builds, I haven't tried to deploy it yet. Do you have any suggestions on the best way to do that before merging?

**Comment by mjpitz** [comment]
Easiest way would be to `make docker` and then deploy one of the `docker` configs.

**Comment by mjpitz** [comment]
That being said, my integration tests do build and deploy the container and waits for it to be healthy

**Comment by mjpitz** [comment]
Ran through a few use cases this morning and things look good I need to look into the output from the integration tests a bit more, but all in all I think this is good to go.

**Comment by arlaneenalra** [comment]
Sorry for the delay in replying. It sounds like your integration tests are already doing what I was thinking about, so this is probably good to go.

**Comment by mjpitz** [comment]
Resolves gh-245

**Pull request merged**
